client/landing/gutenboarding/path.ts

```diff
@@ -97,8 +101,30 @@ export function useNewQueryParam() {
 	return new URLSearchParams( useLocation().search ).has( 'new' );
 }
 
+export function useIsAnchorFm(): boolean {
+	const podcastId = useAnchorFmPodcastId();
+	return podcastId !== null && podcastId !== '';
+}
+
+// Returns the anchor podcast id. First looks in "location state",
+// provided by react-router-dom, if not available there, checks the query string.
+export function useAnchorFmPodcastId(): string | null {
+	const { state } = useLocation< GutenLocationStateType >();
+	const idFromQueryString = useAnchorFmQueryParam();
+	if ( state && state.anchorFmPodcastId ) {
+		return state.anchorFmPodcastId;
+	}
+	return idFromQueryString;
+}
+
 // Returns podcastId value if url has a `?anchor_podcast`, which is used
 // for sites created by Anchor.fm users.
-export function useAnchorFmQueryParam() {
-	return new URLSearchParams( useLocation().search ).get( 'anchor_podcast' ) || '';
+// External callers should prefer useAnchorFmPodcastId(), since navigating steps
+// and pressing f5 can remove the query param.
```

Comment:
Given the potential issues of calling `useAnchorFmQueryParam` directly, should we move this functionality into `useAnchorFmPodcastId`?

Reply:
For this, I removed the `export` so it shouldn't be possible.  However, the current structure does force `useAnchorFmPodcastId()` to do some unnecessary work due to the rule of hooks (because `useAnchorFmQueryParam()` can't be moved to after the state check), so that's an additional argument for inlining.

Reply:
Done here [fix/anchor-gutenboarding-f5 47bcb63b31] Inline useAnchorFmQueryParam()